change-validation/src/main/scala/com/normation/plugins/changevalidation/NotificationService.scala

```diff
@@ -0,0 +1,237 @@
+package com.normation.plugins.changevalidation
+
+import java.io.File
+import java.util.Properties
+
+import bootstrap.liftweb.FileSystemResource
+import com.normation.NamedZioLogger
+import com.normation.plugins.changevalidation.TwoValidationStepsWorkflowServiceImpl.Deployment
+import com.normation.plugins.changevalidation.TwoValidationStepsWorkflowServiceImpl.Validation
+import com.normation.rudder.domain.workflows.WorkflowNode
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import javax.mail.Session
+import javax.mail._
+import javax.mail.internet.InternetAddress
+import javax.mail.internet.MimeMessage
+import zio.IO
+import zio.ZIO
+import zio.syntax._
+
+import scala.util.Success
+import scala.util.Try
+
+case class Email(value: String)
+
+case class Username(value: String)
+
+case class SMTPConf(
+    smtpHostServer : String
+  , port           : Int
+  , email          : Email
+  , login          : Option[Username]
+  , password       : Option[String]
+)
+
+case class EmailInfo(
+  state     : WorkflowNode
+  , to      : Set[Email]
+  , replyTo : Set[Email]
+  , cc      : Set[Email]
+  , bcc     : Set[Email]
+  , subject : String
+  , template: String
+)
+
+
+object NotificationService {
+
+  val path            = "/vagrant/email.conf"
+  val config          = getSMTPConf(path)
+  val validationEmail = getValidationMailParameter(path)
+  val deploymentEmail = getDeploymentMailParameter(path)
+  val logger          = NamedZioLogger("plugin.change-validation")
+
+
+  def sendNotification(text: String, state: WorkflowNode) = {
+    val notif = for {
+      serverConfig <- getSMTPConf(path)
+      mailParameter <- state match {
+        case Validation => getValidationMailParameter(path)
+        case Deployment => getDeploymentMailParameter(path)
+      }
+//      content <- getContentFromTemplate(path)
+      sendingMail <- sendEmail(serverConfig, "toto", mailParameter)
+    } yield (sendingMail)
+
+    notif.catchAll {
+      err =>
+        for {
+          _ <- logger.error(s"Sending `${state.id.value}` notification have failed, cause by : ${err}")
+        } yield {
+          state
+        }
+    }
+  }
+
+  //TO-DO : handle errors with ZIO, exceptions, Task ?
+  def sendEmail(conf: SMTPConf, content: String, mailParameter: EmailInfo) = {
+
+    val prop = new Properties()
```

Comment:
Did you tried zio-email ?

Reply:
The import didn't work, the repo doesn't seems to be active